docs/content/direct/multi-wec-aggregated-status.md

```diff
@@ -0,0 +1,196 @@
+# KubeStellar Multi-WEC Status Aggregation Proposal
+
+**Status:** Design for Review | **Issue:** #2809
+
+---
+
+## 1. Problem Statement
+
+**Current Behavior:**
+
+* Singleton workloads (nWECs == 1) propagate `.status` correctly in WDS objects.
+* Multi-WEC deployments (nWECs > 1) leave WDS `.status` empty, preventing users and tools like ArgoCD from seeing true readiness.
+
+**Goal:**
+
+* Aggregate per-WEC `WorkStatus` and update `.status` in WDS workload objects to reflect **true multi-cluster readiness**.
+* Maintain **backward compatibility** with singleton workloads.
+* Avoid conflicts with other controllers (e.g., ArgoCD, Helm).
+* Ensure **deterministic and observable behavior** across multiple BindingPolicies.
+
+---
+
+## 2. Key Principles
+
+1. **Reuse Existing Patterns**
+
+   * Collect WorkStatus slices per WDS object using the CombinedStatus slice/map approach.
+   * Reuse `updateObjectStatus()` from SingletonStatus for safe `.status` patching.
+
+2. **Predefined Field-Type Aggregation**
+
+   * Rules are **field-type based**, no user-defined expressions initially.
+   * Configurable for future analytics (sum, average, quorum).
+
+3. **Direct WDS Update**
+
+   * Aggregated status is pushed **directly into WDS workload objects**, maintaining a single source of truth.
+
+4. **Opt-In**
+
+   * Aggregation triggered via `wantMultiReportedState: true` per BindingPolicy/StatusCollector.
+   * Singleton behavior remains unchanged for nWECs == 1.
+
+5. **Deterministic Multi-BindingPolicy Handling**
+
+   * Conflicting policies fall back to **conservative defaults** (`min()` for numeric, AND for negative conditions, OR for positive conditions).
+   * `.status.kubestellar/policyBreakdown` records contributing policies and modes for observability.
+
+---
+
+## 3. Slice/Map Data Gathering
+
+```go
+c.workStatusToObject.ReadInverse().ContGet(wObjID, func(wsONSet sets.Set[cache.ObjectName]) {
+    for wsON := range wsONSet {
+        ws := c.workStatusLister.Get(wsON)
+        workStatuses = append(workStatuses, WorkStatusData{
+            ClusterName: ws.ClusterName,
+            Status:      ws.Status,
+            UpdateTime:  ws.LastUpdateTime,
+        })
+    }
+})
+```
+
+* Slice mirrors CombinedStatus:
+  * `.status` map from each WEC
+  * Cluster metadata
+  * Last update timestamp
+* Stale WorkStatus (`LastUpdateTime` older than configurable threshold, e.g., 30s) is ignored during aggregation.
```

Comment:
This is not conservative. A WEC that has not reported in recently is still a WEC.